### DISCUSSION TITLE
Prevent extensions from using subdomain with port number

### DISCUSSION THREAD

**Discussion by themadman0980** [coder/code-server#6407] [General]
I have code-server set up using LSIO, including a custom domain with my reverse proxy (SWAG).

I'm trying to use an extension with code-server, and it's mapping a file view to https://[PORT].code-server.mydomain.duckdns.org. Code-server doesn't like this, as it's forced to use an unsigned cert for this domain, and won't display the file within code-server at all.

Is there any way to have the file mapped to https://code-server.mydomain.duckdns.org:PORT instead? I have a cert covering this.

**Comment by code-asher** [comment]
Are you using `--proxy-domain`?  code-server should only be using a sub-domain with either `--proxy-domain` or `VSCODE_PROXY_URI` set because the default is to use a path-based proxy (in this case `https://code-server.mydomain.duckdns.org/port/[PORT]`).

You can skip code-server's proxy and make it use your own thing by setting the `VSCODE_PROXY_URI` template.  For example:

```
export VSCODE_PROXY_URI=https://code-server.mydomain.duckdns.org:{{port}}
code-server
```

I have never used it with `:` before so I am not 100% sure it will work.  It is possible we need to make a tweak to handle this case.
